trie/proof_test.go

```diff
@@ -451,7 +451,7 @@ func TestBadRangeProof(t *testing.T) {
 			// with the first/last second element(since small values are
 			// embedded in the parent). Avoid this case.
 			index = mrand.Intn(end - start)
-			if (index == end-start-1 || index == 0) && end <= 100 {
+			if (index == 0 && start < 100) || (index == end-start-1 && end <= 100) {
```

Comment:
I don't understand this change. Why does it matter that start is < 100?

Reply:
Because if the gapped element is the first one in the batch. And also the batch contains some small values(embedded nodes). In this case, the edge proof can be used as proof for the second element.

So if we pick the first element for gapping, we need to ensure it's not small value.